src/dos/cdrom_image.cpp

```diff
@@ -552,7 +519,7 @@ bool CDROM_Interface_Image::PlayAudioSector(uint64_t start, uint64_t len)
 	                    + clamp(relative_start, 0, static_cast<int>(track->length - 1))
 	                    * track->sectorSize);
 
-	TrackFile* trackFile = track->file.get();
+	std::shared_ptr<TrackFile> trackFile(track->file);
```

Comment:
Another great catch; you're absolutely right -- the weak_ptr could be killed between the conditional test and its use. 

Regarding the compile-time test:
``` c++
std::shared_ptr<TrackFile> trackFile = track->file;
std::shared_ptr<TrackFile> trackFile(track->file);
```
For my own understanding, I would expect both to behave the same at compile time, if both the assignment / copy-operator's RHS and constructor's argument accept the same type?

Regarding using constructor-instantiation instead of assignment ( (my understand might also be dated) was concerned that for non-trivial types, the assignment `=` instantiation first creates the object with default members which are then overwritten by the RHS (so essentially doubling the work); versus using the constructor `( )` that assigns the values straight away. Maybe C++11 has fixed this?